ceph/ceph.go

```diff
@@ -304,22 +429,28 @@ func createNamespace(ns string) core.Namespace {
 
 // getCephDaemonNames scans the path to ceph sockets in search of an instance which name contains specified prefix and extension.
 // Returns names of available ceph-daemon sockets
-func (s *Socket) getCephDaemonNames() []string {
+func (s *Socket) getCephDaemons() ([]Daemon, error) {
 	files, err := ioutil.ReadDir(s.path)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Not found any ceph-daemon socket in %+v. Check if the given path is correct\n", s.path)
-		panic(err)
+		return nil, err
 	}
-	names := []string{}
+	daemons := []Daemon{}
 	suffix := "." + s.ext
 	for _, f := range files {
 		// read names of all items in socket path with specific prefix-name (defaults to "ceph-") and extension (defaults to .asok)
 		if strings.HasPrefix(f.Name(), s.prefix) && strings.HasSuffix(f.Name(), suffix) {
-			names = append(names, f.Name())
+			daemonNameSplit := strings.Split(trimPrefixAndSuffix(f.Name(), s.prefix, "."+s.ext), ".")
+			daemonName := daemonNameSplit[0]
+			daemonID := ""
+			if len(daemonNameSplit) > 1 {
+				daemonID = daemonNameSplit[1]
+			}
+			daemons = append(daemons, Daemon{id: daemonID, name: daemonName, fullName: f.Name()})
```

Comment:
I have a question - is it possible that daemon full name does not contain daemon ID? As I know this name always contains an ID, so this points to specific daemon, eg. osd.0, osd.1. Please confirm if that is true or not. So, if this full name should contain an ID, the logic here (446-448) should be changed to be more restrict. 

The second aspect is this fragment:
```
daemonName := daemonNameSplit[0]
daemonID := ""
if len(daemonNameSplit) > 1 {
	daemonID = daemonNameSplit[1]
}
daemons = append(daemons, Daemon{id: daemonID, name: daemonName, fullName: f.Name()})
```
Consider changing it to something like below (but it's light suggestion, up to you) :
```
daemon := Daemon{name: daemonNameSplit[0], fullName: f.Name()}
 if len(daemonNameSplit) > 1 {
       //set daemon id
	daemon.id = daemonNameSplit[1]
}
daemons = append(daemons, daemon)
```